cmake/templates/OpenCVConfig.cmake.in

```diff
@@ -46,6 +46,11 @@ if(NOT DEFINED OpenCV_MODULES_SUFFIX)
 endif()
 
 if(NOT TARGET opencv_core)
+  # Extract directory name from full path of the file currently being processed.
+  # Note that CMake 2.8.3 introduced CMAKE_CURRENT_LIST_DIR, but we do this
+  # manually to support older versions of CMake as well.
+  get_filename_component(CMAKE_CURRENT_LIST_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
```

Comment:
It will be good to add CMake version check (to eliminate violation of future CMake policy checks about reserved/read only variables):

```
if(CMAKE_VERSION VERSION_LESS "2.8.3")
```